app/processImage.js

```diff
@@ -41,14 +45,18 @@ function processImage(key, imageParams, processImageCallback) {
                 callback({}); // call with err
             }
         },
+        function(data, fileInfo, callback) {
+            applyBlur(data, imageParams.blur, callback)
+        },
         function(data, fileInfo, callback) {
             // save file to S3
             console.log("Saving file to storage");
             storage.storage.saveFile(key.replace('/',''), data, fileInfo, callback);
-        }
+        },
     ], function(err, data) {
-        // this function is always executed both in case of err and success as well
-        console.log("Error raised while Processing raw File, Error=", err, ", data=", data);
+        if(err) {
```

Comment:
Is there a reason we wrapped this in error. We might not be able to identify this whole block was called or not.

This whole block is the error block  for `waterfall`

Reply:
Its the last callback to be executed once all the functions of waterfall have been completed.
The block is wrapped in error as It should only log when there is actually some king of error.

Reply:
gotcha. Thanks for the explaination 👍